core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -74,6 +74,16 @@ public SharePartitionManager(ReplicaManager replicaManager, Time time, ShareSess
         this.time = time;
         this.cache = cache;
         this.partitionCacheMap = new ConcurrentHashMap<>();
+        fetchQueue = new ConcurrentLinkedQueue<>();
+        this.processFetchQueueLock = new AtomicBoolean(false);
+    }
```

Comment:
Also why not to call the other constructor from public constructor by calling `this(replicaManager, time, cache, new ConcurrentHashMap<>());`